peer/test_utils.go

```diff
@@ -726,3 +728,67 @@ func createTestPeer(t *testing.T) *peerTestCtx {
 		chanStatusMgr: chanStatusMgr,
 	}
 }
+
+func startPeer(t *testing.T, mockConn *mockMessageConn, peer *Brontide) {
+	errChan := make(chan error)
+	go func() {
```

Comment:
I think it may be cleaner if you put the `peer.Start()` in the goroutine and keep this function body synchronous in this function. You can still do the errChan thing for the result of `peer.Start()`